src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -155,51 +167,61 @@ private void FetchSuggestions(bool isAutomaticOnInitialization)
-            try
+            if (IsUsingSemanticContext)
             {
-                var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
-                var renameLocations = await this.BaseViewModel.Session.AllRenameLocationsTask.JoinAsync(cancellationToken)
+                var document = this.BaseViewModel.Session.TriggerDocument;
+                var smartRenameContext = ImmutableDictionary<string, ImmutableArray<(string filePath, string content)>>.Empty;
+                try
+                {
+                    var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
+                    var renameLocations = await this.BaseViewModel.Session.AllRenameLocationsTask.JoinAsync(cancellationToken)
+                        .ConfigureAwait(false);
+                    var context = await editorRenameService.GetRenameContextAsync(this.BaseViewModel.Session.RenameInfo, renameLocations, cancellationToken)
+                        .ConfigureAwait(false);
+                    smartRenameContext = ImmutableDictionary.CreateRange<string, ImmutableArray<(string filePath, string content)>>(
+                        context
+                        .Select(n => new KeyValuePair<string, ImmutableArray<(string filePath, string content)>>(n.Key, n.Value)));
+                }
+                catch (Exception e) when (FatalError.ReportAndCatch(e, ErrorSeverity.Diagnostic))
+                {
+                    // use empty smartRenameContext
+                }
+                _ = await _smartRenameSession.GetSuggestionsAsync(smartRenameContext, cancellationToken)
                     .ConfigureAwait(false);
-                var context = await editorRenameService.GetRenameContextAsync(this.BaseViewModel.Session.RenameInfo, renameLocations, cancellationToken)
-                    .ConfigureAwait(false);
-                smartRenameContext = ImmutableDictionary.CreateRange<string, ImmutableArray<(string filePath, string content)>>(
-                    context
-                    .Select(n => new KeyValuePair<string, ImmutableArray<(string filePath, string content)>>(n.Key, n.Value)));
             }
-            catch (Exception e) when (FatalError.ReportAndCatch(e, ErrorSeverity.Diagnostic))
+            else
             {
-                // use empty smartRenameContext
+                _ = await _smartRenameSession.GetSuggestionsAsync(cancellationToken)
+                    .ConfigureAwait(false);
             }
-            _ = await _smartRenameSession.GetSuggestionsAsync(smartRenameContext, cancellationToken)
-                .ConfigureAwait(false);
         }
-        else
+        finally
         {
-            _ = await _smartRenameSession.GetSuggestionsAsync(cancellationToken)
-                .ConfigureAwait(false);
+            this.IsInPreparation = false;
```

Comment:
note: this will be on a BG thread.  we likely need to come back to the UI for this.